File: app/src/main/java/com/fthdgn/books/callback/implementation/LogComponentCallbacks.java
package com.fthdgn.books.callback.implementation;

import android.content.ComponentCallbacks2;
import android.content.res.Configuration;

import com.fthdgn.books.utils.LogUtils;

public class LogComponentCallbacks implements ComponentCallbacks2 {

    @Override
    public void onTrimMemory(int level) {
        LogUtils.lifecycle(LogComponentCallbacks.this, "onTrimMemory");
    }

    @Override
    public void onConfigurationChanged(Configuration newConfig) {
        LogUtils.lifecycle(LogComponentCallbacks.this, "onConfigurationChanged " + newConfig.toString());
    }

    @Override
    public void onLowMemory() {
        LogUtils.lifecycle(LogComponentCallbacks.this, "onLowMemory");
    }
}

File: app/src/main/java/com/fthdgn/books/callback/implementation/LogComponentCallbacks.kt
package com.fthdgn.books.callback.implementation

import android.content.ComponentCallbacks2
import android.content.res.Configuration

import com.fthdgn.books.utils.LogUtils

class LogComponentCallbacks : ComponentCallbacks2 {

    override fun onTrimMemory(level: Int) {
        LogUtils.lifecycle(this@LogComponentCallbacks, "onTrimMemory")
    }

    override fun onConfigurationChanged(newConfig: Configuration) {
        LogUtils.lifecycle(this@LogComponentCallbacks, "onConfigurationChanged " + newConfig.toString())
    }

    override fun onLowMemory() {
        LogUtils.lifecycle(this@LogComponentCallbacks, "onLowMemory")
    }
}
